CMakeLists.txt

```diff
@@ -58,7 +59,35 @@ else()
     option(QD_ENABLE_ASSERTIONS "Enable assertions" OFF)
 endif()
 
-file(STRINGS "${CMAKE_SOURCE_DIR}/VERSION.txt" QPID_DISPATCH_VERSION)
+if(NOT DEFINED VERSION)
+  # Version was not provided. First check to see if git is avaialble
+  find_package(Git)
+   if(Git_FOUND)
+      execute_process(COMMAND ${GIT_EXECUTABLE} describe --tags --dirty=-modified --always OUTPUT_VARIABLE DEFAULT_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+      # You might sometimes get a fatal error when running the above command (when this is not a git repo).
+      string(FIND ${DEFAULT_VERSION} "fatal" POS)
+      if ('${POS}' STREQUAL '-1')
+         # 'fatal' was not found in the output, we will use the DEFAULT_VERSION as the QPID_DISPATCH_VERSION
+         set(QPID_DISPATCH_VERSION ${DEFAULT_VERSION})
+       else()
+         set(QPID_DISPATCH_VERSION "UNKNOWN")
+       endif('${POS}' STREQUAL '-1')
+   else(Git_FOUND)
+       # Git executable was not available, we will not be able to determine the version, just set it to "UNKNOWN"
+       set(QPID_DISPATCH_VERSION "UNKNOWN")
+   endif(Git_FOUND)
```

Comment:
I looked through CI and I saw, in the RPM job, this

```
fatal: not a git repository (or any of the parent directories): .git
-- Found Git: /usr/bin/git (found version "2.31.1") 
CMake Error at CMakeLists.txt:68 (string):
  string sub-command FIND requires 3 or 4 parameters.
```

At the same time, i was considering maybe it would be better to look at the exit code of git, instead of looking for `fatal` in stderr

```suggestion
      execute_process(
              COMMAND ${GIT_EXECUTABLE} describe --tags --dirty=-modified --always
              OUTPUT_VARIABLE DEFAULT_VERSION
              RESULT_VARIABLE GIT_RESULT
              OUTPUT_STRIP_TRAILING_WHITESPACE)
      # You might sometimes get a fatal error when running the above command (when this is not a git repo).
      if (GIT_RESULT EQUAL 0)
         # Git succeeded, we will use the DEFAULT_VERSION as the QPID_DISPATCH_VERSION
         set(QPID_DISPATCH_VERSION ${DEFAULT_VERSION})
       else()
         set(QPID_DISPATCH_VERSION "UNKNOWN")
       endif(GIT_RESULT EQUAL 0)
   else(Git_FOUND)
       # Git executable was not available, we will not be able to determine the version, just set it to "UNKNOWN"
       set(QPID_DISPATCH_VERSION "UNKNOWN")
   endif(Git_FOUND)
```

Reply:
RPM job seems to be doing the right thing now,

```
-- Found Git: /usr/bin/git (found version "2.31.1") 
fatal: not a git repository (or any of the parent directories): .git
-- Setting skupper-router version to UNKNOWN
```

Reply:
Sounds good. Thanks for testing.